### PR TITLE
Changes in LaaP dashboard

### PR DESCRIPTION
**Pull request by SherrinRaji**
This PR includes the below changes:

1. License view list to navigate to filtered Plans within the list.
2. License view lists include icons indicating license status.
3. In-page add/edit for License and mapping to plan and feature models
4. Filters for each view list.

### CODE REVIEW

File: core/admin.py [hunk 1]
```diff
@@ -1,10 +1,45 @@
...
+            + urlencode({"license_id": f"{obj.license_id}"})
+        )
+        return format_html('<a href="{}">Plans</a>', url)
+    view_plans_link.short_description = 'Linked Plans'
+
+@admin.register(LicensePlan)
+class LicensePlanAdmin(admin.ModelAdmin):
+    list_display = ('display_license', 'display_plan')
+    list_filter = ('license_id',)
+admin.register(Plan)
```
Comment (addu390) [inline core/admin.py:39]: Move this to the top (All the register(s) in one place)?

File: core/models.py [hunk 2]
```diff
@@ -27,6 +27,9 @@ class License(BaseModel):
     class Meta:
         db_table = LICENSE
         verbose_name_plural = LICENSE
+    def license_status(self):
+        return self.status == 'ACTIVE'
```
Comment (addu390) [inline core/models.py:31]: Use the constant in `constants.py`

File: core/models.py [hunk 3]
```diff
@@ -70,9 +73,12 @@ class LicensePlan(BaseModel):
     class Meta:
         db_table = "license_plan"
         verbose_name_plural = "License and Plan"
-
-    def __str__(self):
-        return "%s -> %s" % (str(self.license), str(self.plan))
+    def display_plan(self):
+        return str(self.plan)
+    display_plan.short_description = 'Plan'
+    def display_license(self):
```
Comment (addu390) [inline core/models.py:79]: Follow the PEP8 spacing convention for the entire file.

File: core/models.py [hunk 4]
```diff
@@ -97,6 +103,12 @@ class LicenseFeature(BaseModel):
     class Meta:
         db_table = "license_feature"
         verbose_name_plural = "License and Feature"
-
-    def __str__(self):
```
Comment (addu390) [inline core/models.py:101]: Does this look bad on the dashboard?

**Comment by addu390** [comment]
@SherrinRaji, I have reviewed the PR. Can you also add screenshot(s) here in the chat?

**Review by addu390 — CHANGES_REQUESTED** [review]
- PEP8 Formatting.
- Screenshot(s) of the screens.
- Use constants.py wherever necessary.

File: core/admin.py (hunk 1 above)
Comment (SherrinRaji) [reply core/admin.py:39]: Okay sounds good!

File: core/models.py (hunk 4 above)
Comment (SherrinRaji) [reply core/models.py:101]: Yep

**Comment by SherrinRaji** [comment]
I have made the changes according to your comments. 
Screenshot 2022-03-11 at 20 57 53
Screenshot 2022-03-11 at 20 58 15

File: core/admin.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 from gc import get_objects
 from django.contrib import admin
 from django.forms.models import BaseInlineFormSet
-from django import forms
+from django.db.models import *
```
Comment (addu390) [inline core/admin.py:4]: Avoid using `*` import(s), import what's necessary.

File: core/constants.py [hunk 6]
```diff
@@ -1,3 +1,5 @@
+# General constants
+ACTIVE_STATUS = 'ACTIVE'
```
Comment (addu390) [inline core/constants.py:2]: There's already a constant called `ACTIVE `, why is this required?

**Review by addu390 — CHANGES_REQUESTED** [review]
LGTM, remove `ACTIVE_STATUS` in `constant.py` and import `ACTIVE` instead.

File: core/models.py [hunk 7]
```diff
@@ -28,6 +28,11 @@ class Meta:
         db_table = LICENSE
         verbose_name_plural = LICENSE
 
+    def license_status(self):
```
Comment (addu390) [inline core/models.py:31]: What's this for?

File: core/models.py [hunk 8]
```diff
@@ -44,6 +49,9 @@ class Meta:
         db_table = PLAN
         verbose_name_plural = PLAN
 
+    def display_plan(self):
```
Comment (addu390) [inline core/models.py:52]: What's this for?

File: core/models.py (hunk 3 above)
Comment (addu390) [reply core/models.py:88]: PEP8 is not used again.

File: core/models.py [hunk 9]
```diff
@@ -83,6 +102,17 @@ class Meta:
         db_table = "feature_plan"
         verbose_name_plural = "Feature and Plan"
 
+
+    def display_feature(self):
```
Comment (addu390) [inline core/models.py:106]: Looks like a lot of code duplicate for display, what is this for in the Admin Panel?

**Review by addu390 — CHANGES_REQUESTED** [review]
Code duplication?

**Review by addu390 — APPROVED** [review]
LGTM

**Pull request merged**